system/lib/wasmfs/file.cpp

```diff
@@ -40,33 +40,180 @@ void DataFile::Handle::preloadFromJS(int index) {
 // Directory
 //
 
+void Directory::Handle::finishInsertingChild(const std::string& name,
+                                             std::shared_ptr<File> child) {
+  // Update the dcache and set the child's parent.
+  auto& dcache = getDir()->dcache;
+  auto [_, inserted] = dcache.insert({name, {DCacheKind::Normal, child}});
+  assert(inserted && "inserted child already existed!");
+  assert(child->locked().getParent() == nullptr);
+  child->locked().setParent(getDir());
+}
```

Comment:
How about calling this `cacheChild` as it basically adds it to the dcache? Or `dcacheChild`?

We could have such a function take the DCacheKind as a parameter, and then `mountChild` and this function here would call that with different kinds, and we'd have a single place that adds to the dcache, which seems nice.

Reply:
That sounds good, too. `cacheChild` doesn't completely capture the fact that the parent pointer is also updated, but it still communicates the big picture.